src/MPAGSCipher/PlayfairCipher.cpp

```diff
@@ -0,0 +1,171 @@
+#include "PlayfairCipher.hpp"
+
+#include <iostream>
+#include <string>
+#include <algorithm>
+#include <iterator>
+
+PlayfairCipher::PlayfairCipher(const std::string& key){
+    setKey(key);
+}
+
+std::string encounteredletters{""};
+
+void PlayfairCipher::setKey(const std::string& key){
+    // store the original key
+    key_ = key;
+
+    // Append the alphabet
+    key_.append("abcdefghijklmnopqrstuvwxyz");
+
+    // Make sure the key is upper case
+    
+    std::transform(key_.begin(),key_.end(),key_.begin(),toupper);   //transform algorithm which transforms all char in string to upper case
+
+    // Remove non-alpha characters
+    auto iter = std::remove_if(key_.begin(),key_.end(),[](char& val){return !std::isalpha(val);});
+    key_.erase(iter,key_.end());
+    
+    // Change J -> I
+    std::transform(key_.begin(),key_.end(),key_.begin(),[](char& val){if(val=='J'){return 'I';} else {return val;}});
+    
+    // Remove duplicated letters
+    auto iter2 = std::remove_if(key_.begin(),key_.end(),[=](char& val){
+        if(encounteredletters.find(val)!=std::string::npos){
+            return true;
+        } else {
+            encounteredletters += val; 
+            return false;
+        }});
+    key_.erase(iter2,key_.end());
+
+    // Store the coords of each letter and the playfair cipher key map
+
+    for(std::size_t i{0};i<key_.size();i++){
+        Str2CoordMap::value_type invcoords{std::pair<int,int> {(i/5)+1,(i%5)+1},key_[i]};
+        Str2CharMap::value_type coords{key_[i],std::pair<int,int> {(i/5)+1,(i%5)+1}};
```

Comment:
Not making the coords zero indexed (i.e. yours start from 1) is unnecessary and makes the code a bit more tricky to understand when applying the cipher. Because C++ is a zero-indexed language, I would very much recommend sticking to that in your code where possible!